plugins/inputs/dpdk/dpdk.go

```diff
@@ -96,108 +84,206 @@ func (dpdk *dpdk) Init() error {
 
 	dpdk.ethdevExcludedCommandsFilter, err = filter.Compile(dpdk.EthdevConfig.EthdevExcludeCommands)
 	if err != nil {
-		return fmt.Errorf("error occurred during filter prepation for ethdev excluded commands: %w", err)
+		return fmt.Errorf("error occurred during filter preparation for ethdev excluded commands: %w", err)
+	}
+
+	if err = choice.Check(dpdk.UnreachableSocketBehavior, []string{unreachableSocketBehaviorError, unreachableSocketBehaviorIgnore}); err != nil {
+		return fmt.Errorf("unreachable_socket_behavior: %w", err)
+	}
+
+	glob, err := prepareGlob(dpdk.SocketPath)
+	if err != nil {
+		return err
+	}
+	dpdk.socketGlobPath = glob
+
+	if err = dpdk.maintainConnections(); err != nil {
+		if dpdk.UnreachableSocketBehavior == unreachableSocketBehaviorError {
+			return err
+		}
+		dpdk.Log.Warnf("Unreachable socket issue occurred: %v", err)
+	}
+
+	return nil
+}
+
+// Start is empty to implement ServiceInput interface
+func (dpdk *dpdk) Start(telegraf.Accumulator) error {
+	return nil
+}
+
+func (dpdk *dpdk) Stop() {
+	var err error
+	for _, connector := range dpdk.connectors {
+		if err = connector.tryClose(); err != nil {
+			dpdk.Log.Warnf("Couldn't close connection for %q: %v", connector.pathToSocket, err)
+		}
+	}
+	dpdk.connectors = nil
+}
+
+// Gather function gathers all unique commands and processes each command sequentially
+// Parallel processing could be achieved by running several instances of this plugin with different settings
+func (dpdk *dpdk) Gather(acc telegraf.Accumulator) error {
+	if err := dpdk.maintainConnections(); err != nil {
+		if dpdk.UnreachableSocketBehavior == unreachableSocketBehaviorError {
```

Comment:
```suggestion
		if errors.Is(dpdk.UnreachableSocketBehavior, unreachableSocketBehaviorError) {
```

Reply:
@Hipska I believe that it will not compile (both `dpdk.UnreachableSocketBehavior` and `unreachableSocketBehaviorError` are strings based on provided configuration option)

Reply:
Oh sorry, I was getting confused by the variable naming..